src/main/java/org/folio/circulation/resources/context/RenewalContext.java

```diff
@@ -0,0 +1,35 @@
+package org.folio.circulation.resources.context;
+
+import org.folio.circulation.domain.ItemStatus;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.RequestQueue;
+import org.joda.time.DateTimeZone;
+
+import io.vertx.core.json.JsonObject;
+import lombok.Value;
+import lombok.With;
+import lombok.val;
+
+@Value
+@With
+public class RenewalContext {
+  Loan loan;
+  RequestQueue requestQueue;
+  DateTimeZone timeZone;
+  Loan loanBeforeRenewal;
+  ItemStatus itemStatusBeforeRenewal;
+  boolean lostItemFeesRefundedOrCancelled;
+  String loggedInUserId;
+  JsonObject renewalRequest;
+
+  public static RenewalContext newRenewalContext(Loan loan, JsonObject renewalRequest,
```

Comment:
Could `from` or `create` be alternatives to this name?

If there isn't a good name, could this be changed to a constructor instead?

```suggestion
  public static RenewalContext from(Loan loan, JsonObject renewalRequest,
```

```suggestion
  public static RenewalContext create(Loan loan, JsonObject renewalRequest,
```

Reply:
renamed to `create`